zcash_transparent/src/bundle.rs

```diff
@@ -258,6 +266,29 @@ impl TxIn<Authorized> {
     }
 }
 
+impl TxIn<builder::Coinbase> {
+    /// Creates an input for a coinbase transaction. Does not support creating inputs for the
+    /// genesis block.
+    pub fn coinbase(
+        height: BlockHeight,
+        miner_data: &MinerData,
+        sequence: u32,
+    ) -> Result<Self, coinbase::Error> {
+        let height = i64::from(height);
+
+        // Serialize the height. The serialization is specified in
+        // <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>.
+        let mut script_sig = match height {
+            0 => Err(coinbase::Error::GenesisInputNotSupported)?,
+            _ => zcash_script::pattern::push_num(height).to_bytes(),
```

Comment:
This is great code re-use.

No changes needed here, just curious: Does anyone know why zcashd uses a sequence of `u32::MAX` for v4 transactions?